File: src/main/java/net/tassia/calc/value/FunctionCall.java
package net.tassia.calc.value;

public class FunctionCall implements NumericValue {

    private final String function;
    private final NumericValue[] arguments;

    public FunctionCall(String function, NumericValue...arguments) {
        this.function = function;
        this.arguments = arguments;
    }

    public String getFunction() {
        return function;
    }

    public NumericValue[] getArguments() {
        return arguments;
    }

}

File: src/main/java/net/tassia/calc/element/FunctionCall.kt
package net.tassia.calc.element

import net.tassia.calc.ICalculatable
import net.tassia.calc.function.Function

/**
 * Represents a call to a function.
 *
 * @param function the called function
 * @param args the arguments
 *
 * @since Calc 1.0
 * @author Tassilo
 */
class FunctionCall(val function: Function, val args: List<ICalculatable>) : ICalculatable {

	override fun toFunctionString(): String {
		return function.name + "(" + args.joinToString(transform = ICalculatable::toFunctionString) + ")"
	}

	override fun evaluateDouble(): Double {
		return function(args).evaluateDouble()
	}

	override fun evaluateLong(): Long {
		return function(args).evaluateLong()
	}

}
